code/game/objects/effects/meteors.dm

```diff
@@ -112,36 +153,70 @@ GLOBAL_LIST_INIT(meteors_gore, list(/obj/effect/meteor/meaty = 5, /obj/effect/me
 /obj/effect/meteor/Process_Spacemove(movement_dir, continuous_move)
 	return TRUE
 
-/obj/effect/meteor/Bump(atom/A)
-	if(A)
-		ram_turf(get_turf(A))
-		playsound(loc, meteorsound, 40, TRUE)
-		if(!istype(A, /obj/structure/railing))
-			get_hit()
-
-/obj/effect/meteor/proc/ram_turf(turf/T)
-	//first bust whatever is in the turf
-	for(var/thing in T)
-		var/atom/A = thing
-		if(thing == src)
+/obj/effect/meteor/proc/ram_wall(turf/simulated/wall/wall)
+	var/damage_cap = wall.damage_cap
+	if(wall.rotting)
+		damage_cap /= 10
+	var/damage_needed = (damage_cap - wall.damage) * OBJ_INTEGRITY_TO_WALL_DAMAGE 
+	if(damage_needed <= 0)
+		wall.dismantle_wall()
+	else if(damage_needed < obj_integrity)
+		obj_integrity -= damage_needed
+		wall.dismantle_wall()
+	else
+		wall.take_damage(obj_integrity / OBJ_INTEGRITY_TO_WALL_DAMAGE)
+		if(wall) // In case we somehow broke it despite not thinking we could.
+			wall.ex_act(explosion_strength)
+		deconstruct(FALSE)
+
+/obj/effect/meteor/proc/ram_turf_contents(turf/T)
+	. = FALSE // Tracks whether we hit anything.
+
+	for(var/atom/thing in T)
```

Comment:
```suggestion
	for(var/atom/thing as anything in T)
```

Reply:
Also it may allow nulls to slip in, its probably best to keep it as not anything